### ISSUE TITLE
Try pinging servers in the "Servers" tab

### ISSUE THREAD

**Issue by ghost** [PrismLauncher/PrismLauncher#271]
### Role

I use the launcher.

### Suggestion

I want there to be an option for the launcher to ping servers in an instance's Servers for their MOTDs and playercounts.

### Benefit

It would look cool and it may be useful in case you want to see if it's even worth starting up the instance to play on a server

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

**Comment by MMK21Hub** [comment]
Documentation for the server pinging protocol: https://wiki.vg/Server_List_Ping
It has an example written in C, but not C++.

**Comment by ghost** [comment]
It should only ping if you hit a button like `Refresh` and cache the data somewhere

**Comment by ghost** [comment]
that is a good idea

**Comment by TheAirBlow** [comment]
> It should only ping if you hit a button like `Refresh` and cache the data somewhere

The whole purpose of it so you can see if it's even worth to start Minecraft. You don't need outdated data for that. You can just ping ONLY when you open the tab,.

**Comment by ghost** [comment]
> > It should only ping if you hit a button like `Refresh` and cache the data somewhere
> 
> The whole purpose of it so you can see if it's even worth to start Minecraft. You don't need outdated data for that. You can just ping ONLY when you open the tab,.

or just hit the `Refresh` button....

**Comment by TheAirBlow** [comment]
> > > It should only ping if you hit a button like `Refresh` and cache the data somewhere
> > 
> > 
> > The whole purpose of it so you can see if it's even worth to start Minecraft. You don't need outdated data for that. You can just ping ONLY when you open the tab,.
> 
> or just hit the `Refresh` button....

The problem is that it would be just harder to code because of caching. Also, don't ignore the fact that caching dynamic information is super dumb.

**Comment by ghost** [comment]
> > > > It should only ping if you hit a button like `Refresh` and cache the data somewhere
> > > 
> > > 
> > > The whole purpose of it so you can see if it's even worth to start Minecraft. You don't need outdated data for that. You can just ping ONLY when you open the tab,.
> > 
> > 
> > or just hit the `Refresh` button....
> 
> The problem is that it would be just harder to code because of caching. Also, don't ignore the fact that caching dynamic information is super dumb.

yeah true. but still keep a refresh button so you don't have to reopen the tab everytime

**Comment by TheAirBlow** [comment]
> > > > > It should only ping if you hit a button like `Refresh` and cache the data somewhere
> > > > 
> > > > 
> > > > The whole purpose of it so you can see if it's even worth to start Minecraft. You don't need outdated data for that. You can just ping ONLY when you open the tab,.
> > > 
> > > 
> > > or just hit the `Refresh` button....
> > 
> > 
> > The problem is that it would be just harder to code because of caching. Also, don't ignore the fact that caching dynamic information is super dumb.
> 
> yeah true. but still keep a refresh button so you don't have to reopen the tab everytime

I never said to remove it lol

**Comment by fizzzydev** [comment]
Just wanted to suggest that. MCCI had a maintenance and I wanted to check if the server is still unjoinable or back.

**Comment by iTrooz** [comment]
> Documentation for the server pinging protocol: [wiki.vg/Server_List_Ping](https://wiki.vg/Server_List_Ping) It has an example written in C, but not C++.

Just made one: https://github.com/iTrooz/Cpp-Qt-Minecraft-Ping

**Comment by Trial97** [comment]
fixed by #3112
